### PR TITLE
WT-9142 Deprecate cursor prefix_search configuration and remove related code.

### PR DESCRIPTION
**Pull request by luke-pearson**
Finishing up the bounded cursor project we should remove prefix search from the codebase to tidy things up.

### CODE REVIEW

File: src/btree/bt_cursor.c [hunk 1]
```diff
@@ -1050,32 +1037,20 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
      */
     if (valid) {
         exact = cbt->compare;
```
Comment (luke-pearson) [inline src/btree/bt_cursor.c:1039]: This is the only place where the deletion wasn't straightforward, it requires additional review.

File: src/btree/bt_cursor.c [hunk 2]
```diff
@@ -1050,32 +1037,20 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
...
-                    exact = cbt->recno < state.recno ? -1 : cbt->recno == state.recno ? 0 : 1;
-            }
-            goto done;
+        if (WT_CURSOR_BOUNDS_SET(cursor)) {
+            if (btree->type == BTREE_ROW)
+                WT_ERR(__wt_compare(session, btree->collator, &cursor->key, &state.key, &exact));
+            else
+                exact = cbt->recno < state.recno ? -1 : cbt->recno == state.recno ? 0 : 1;
         }
+        goto done;
```
Comment (jiechenbo) [inline src/btree/bt_cursor.c:1053]: We can remove the goto statement here, previously we would check if prefix matched between the current key and the prefix key. If they did not match we would force it to go into the `cursor_search_neighbouring` call. Now we can remove the goto statement, and inside connect the if statement onto L1081.

**Review by clarissecheah — APPROVED** [review]
Thanks for cleaning this up. LGTM.

File: src/include/btree_cmp_inline.h [hunk 3]
```diff
@@ -127,16 +127,6 @@ __wt_compare(WT_SESSION_IMPL *session, WT_COLLATOR *collator, const WT_ITEM *use
...
 }
 
-/*
- * __wt_prefix_match --
- *     Check if the prefix item is equal to the leading bytes of the tree item.
- */
-static inline int
-__wt_prefix_match(const WT_ITEM *prefix, const WT_ITEM *tree_item)
-{
-    return (__wt_lex_compare(prefix, tree_item, true));
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply src/include/btree_cmp_inline.h:137]: a8101d903

File: src/btree/bt_cursor.c (hunk 2 above)
Comment (luke-pearson) [reply src/btree/bt_cursor.c]: 668ac19e2

**Review by jiechenbo — APPROVED** [review]
LGTM!

**Pull request merged**